Opentype/Fileformat/Cmap.hs

```diff
@@ -166,16 +165,15 @@ readCmap bs_ = do
   c <- index16 bs_ 0
   let bs | (c >= 8 && c < 14) = Strict.drop 8 bs_
          | otherwise = Strict.drop 4 bs_
-  either fail return $
-    case c of 
-      0 -> getMap0 bs
-      2 -> getMap2 bs
-      4 -> getMap4 bs
-      6 -> getMap6 bs
-      8 -> getMap8 bs
-      10 -> getMap10 bs
-      12 -> getMap12 bs
-      i -> fail $ "unsupported map encoding " ++ show i
+  case c of
+    0 -> getMap0 bs
+    2 -> getMap2 bs
+    4 -> getMap4 bs
+    6 -> getMap6 bs
+    8 -> getMap8 bs
+    10 -> getMap10 bs
+    12 -> getMap12 bs
+    i -> Left $ "unsupported map encoding " ++ show i
```

Comment:
1. Eliminating a couple more uses of `fail`.

2. The `case` expression is already in `Either String` so I’ve removed the conversion & dedented.